qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/customnavview/MapboxNavigationViewCustomizedActivity.kt

```diff
@@ -306,6 +323,19 @@ class MapboxNavigationViewCustomizedActivity : DrawerActivity() {
         }
     }
 
+    private fun toggleOnMapLongClick(enable: Boolean) {
+        // If enabled [NavigationView] will intercept map long clicks
+        binding.navigationView.customizeViewOptions {
+            enableMapLongClickIntercept = enable
+            val gesturesPlugin = binding.navigationView.api.getMapView()?.gestures
```

Comment:
Best to use the `onMapViewAttached` and `onMapViewDetached` here because this does not work with the "use custom map" toggle.

Reply:
``` kotlin
private val customMapLongClick = object : NavigationViewListener(), OnMapLongClickListener {

    override fun onMapViewAttached(mapView: MapView) {
        mapView.gestures.addOnMapLongClickListener(this)
    }

    override fun onMapViewDetached(mapView: MapView) {
        mapView.gestures.removeOnMapLongClickListener(this)
    }

    override fun onMapLongClick(point: Point): Boolean {
        Toast.makeText(
            this@MapboxNavigationViewCustomizedActivity,
            "Long press handled by activity",
            Toast.LENGTH_LONG
        ).show()
        return false
    }
}

private fun toggleOnMapLongClick(enable: Boolean) {
    if (enable) {
        binding.navigationView.removeListener(customMapLongClick)
    } else {
        binding.navigationView.addListener(customMapLongClick)
    }
    binding.navigationView.customizeViewOptions {
        enableMapLongClickIntercept = enable
    }
}
```